doc/design/api.md

```diff
@@ -0,0 +1,90 @@
+# Design Doc: PaddlePaddle API
+
+## Ingredients
+
+As the first step of our design, we list important concepts in deep
+learning and try to figure their relationship, as shown below:
+
+```
+Model = {topology, parameters}
```

Comment:
I think "creating topology using Python and run training using C++/Go" is not a common case, but due to that Tensorflow doesn't provide a language binding that works both good for specifying topology and for training. In other words, I don't see a chance that we need to load only topology, or only parameters, but not both.